components/gitpod-protocol/src/workspace-instance.ts

```diff
@@ -27,12 +27,15 @@ export interface WorkspaceInstance {
     stoppedTime?: string;
 
     // ideUrl is the URL at which the workspace is available on the internet
+    // Note: this is nitially empty, filled during starting process!
```

Comment:
IMO this should be an optional attribute then.

Reply:
Definitely. This variance was a side-effect of the introduction of `ws-manager`, and has been this way ever since. Won't change that in the scope of this PR.